app/api/users/userDetails/route.ts

```diff
@@ -1,13 +1,16 @@
-import { getDataFromToken } from "@/helpers/getDataFromToken";
 import { connectMongoDB } from "@/dbConfig/dbConfig";
 import { NextResponse, NextRequest } from "next/server";
+import { getDataFromHeader } from "@/helpers/getDataFromHeader";
 import User from "@/models/userModel";
 
 connectMongoDB();

```

Comment:
To bym przeniósł do try catcha wewnątrz GET'a, może się zdarzyć tak że połączenie się nie uda, wtedy zwrócimy w catchu jakiś błąd.